controllers/chat/group/changeChannelDetail.js

```diff
@@ -27,6 +27,15 @@ const changeChannelDetail = async (req, res) => {
   if (channel_name) updateData.name = channel_name;
 
   try {
+    await channel.sendMessage({
+      text: 'Update Info group',
+      type: 'system',
+      user_id: userId,
+      updatedInfo: {
+        ...updateData
+      }
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code sends a system message before the channel is actually updated. If the `channel.updatePartial` operation fails for any reason, the system message would have been sent prematurely. Consider moving the `sendMessage` call after the `updatePartial` call to ensure that the update was successful before sending the notification.
```diff
-    await channel.sendMessage({
-      text: 'Update Info group',
-      type: 'system',
-      user_id: userId,
-      updatedInfo: {
-        ...updateData
-      }
-    });
-
-    const response = await channel.updatePartial({
-      set: {
-        ...updateData,
+    const response = await channel.updatePartial({
+      set: {
+        ...updateData,
+    });
+
+    await channel.sendMessage({
+      text: 'Update Info group',
+      type: 'system',
+      user_id: userId,
+      updatedInfo: {
+        ...updateData
+      }
     });
```
Also, consider adding error handling around these operations to handle any potential exceptions and provide meaningful feedback to the user.


<!-- This is an auto-generated comment by OSS CodeRabbit -->